VEDA_OS01/tests/test_views.py

```diff
@@ -0,0 +1,232 @@
+            missing_keys
+        )
+
+    @data(
+        {
+            'org': 'test',
+            'provider': TranscriptProvider.CIELO24,
+            'api_key': 'test-api-key',
+            'username': 'test-cielo-user'
+        },
+        {
+            'org': 'test',
+            'provider': TranscriptProvider.THREE_PLAY,
+            'api_key': 'test-api-key',
+            'api_secret_key': 'test-secret-key'
+        }
+    )
+    @responses.activate
+    @patch('VEDA_OS01.views.LOGGER')
+    def test_transcript_credentials_success(self, post_data, mock_logger):
+        """
+        Test that post credentials works as expected.
+        """
+        # Mock get_cielo_token_mock to return token
+        responses.add(
+            responses.GET,
+            CIELO24_LOGIN_URL,
+            body='{"ApiToken": "cielo-api-token"}',
+            status=status.HTTP_200_OK
+        )
+
+        # Verify that transcript credentials are not present for this org and provider.
+        transcript_credentials = TranscriptCredentials.objects.filter(
+            org=post_data.get('org'),
+            provider=post_data.get('provider')
+        )
+        self.assertFalse(transcript_credentials.exists())
+
+        response = self.client.post(self.url, data=json.dumps(post_data), content_type='application/json')
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        transcript_credentials = TranscriptCredentials.objects.filter(
+            org=post_data.get('org'),
+            provider=post_data.get('provider')
+        )
+        self.assertTrue(transcript_credentials.exists())
+
+    @patch('VEDA_OS01.views.LOGGER')
+    @responses.activate
+    def test_cielo24_error(self, mock_logger):
+        """
+        Test that when invalid cielo credentials are supplied, we get correct response.
+        """
+        # Mock get_cielo_token_response.
+        error_message = 'Invalid transcript credentials supplied'
+        responses.add(
+            responses.GET,
+            CIELO24_LOGIN_URL,
+            body=json.dumps({'error': error_message}),
+            status=status.HTTP_400_BAD_REQUEST
+        )
+
+        post_data = {
+            'org': 'test',
+            'provider': TranscriptProvider.CIELO24,
+            'api_key': 'test-api-key',
+            'username': 'test-cielo-user',
+            'api_secret_key': ''
+        }
+        response = self.client.post(
+            self.url,
+            data=json.dumps(post_data),
+            content_type='application/json'
+        )
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        response = json.loads(response.content)
+        self.assertEqual(response['message'], error_message)
+        mock_logger.warning.assert_called_with(
+            '[Transcript Credentials] Unable to get api token --  response %s --  status %s.',
+            json.dumps({'error': error_message}),
+            status.HTTP_400_BAD_REQUEST
+        )
```

Comment:
I am not sure what exactly we are testing in this test. cielo post data is correct, so why we are considering it invalid?

Reply:
Here we are testing when invalid credentials are provided to get cielo token